### DISCUSSION TITLE
strange behavior (bug?)

### DISCUSSION THREAD

**Discussion by ThomasMeister123** [osmandapp/OsmAnd#21306] [Q&A]
I've noticed the following recently: After navigation starts normally (with correct announcements), the announcements suddenly stop. The next instruction is a point very close to the destination (top left), although there are many exits and junctions up to that point. I then have to stop the navigation and start it again with the same destination. Then the next instruction is correct (top left) and the announcements are correct again.

That's why I've already missed an exit because it wasn't announced.

Example 1:
before restart routing:
![A1](https://github.com/user-attachments/assets/439dfc26-7619-4f00-b793-b284a525a6ec)
restart routing:
![A2](https://github.com/user-attachments/assets/03eb88d5-355b-4b07-86ad-10383469316b)

Example 2:
before restart routing:
![B1](https://github.com/user-attachments/assets/5c0389da-b094-4724-8db3-bdab63bd6b7c)

restart routing:
![B2](https://github.com/user-attachments/assets/1105a77b-f406-4e21-9ae1-ff2677a59f42)

**Comment by DmitryAlexei** [comment]
The same issue here 
- https://github.com/osmandapp/OsmAnd/issues/22085

**Comment by ThomasMeister123** [comment]
I can reproduce it. Its ALWAYS at the same route I'm driving multiple times a year.
I have to stop navigation and enter the destinatiuon again (while driving), then everything wrks fine.

**Comment by Zirochkabila** [comment]
@ThomasMeister123 Please specify the start and finish (coordinates) of your route - we will try to reproduce it

**Comment by ThomasMeister123** [comment]
start: 50.04399352952249, 8.204658067338682
finish: 50.29166319086929, 6.671794237925299

Problem starts at autobahn near: 49.97956896050781, 8.006453954475095

At this position, it shows the next "turn" is in 133km. When I restart navigation (not the app, just the routing) it shows the correct next turn at 69km ("Kreuz Koblenz")

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1d7bc0fd-3dd0-4aba-9757-e0db2fedfb21"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by Zirochkabila** [comment]
`OsmAnd~ 5.1.0#5364m, Released: 2025-06-11`

In the laboratory, it was not possible to reproduce this either through location simulation or navigation simulation.
I will observe and investigate real trips
